### DISCUSSION TITLE
Remove all types which doctrine dbal does not support

### DISCUSSION THREAD

**Discussion by Temepest74** [laravel/framework#48227] [Ideas] [answered]
I know this is a long lasting battle, where Laravel teams blame the dbal team, and vice versa. But I think Laravel should remove types such as tiny int and every other type not supported by doctrine dbal. 

I feel like they exists only to confuse new people to coming to Laravel. Like, fine, I need a small data type, I use tiny int, now I need to rename it, oh cool, I MUST change the type to short int (or smth else), then rename it, then leave it as short int cuz why bother with the process again?

**Comment by henzeb** [comment]
Or under the hood use the next best choice. So I don't have to remember that a varchar is a string in laravel....

**Comment by hafezdivandari** [comment]
This is already fixed on v9.44:

- https://github.com/laravel/framework/pull/45258
- https://laravel-news.com/laravel-9-44-0
- https://github.com/laravel/framework/pull/45487

You have native renaming and dropping since 9.44 and native modifying since Laravel 10.

**Reply by Temepest74** [reply]
Is this anywhere inside the docs? I cant find it

**Reply by hafezdivandari** [reply, marked as answer]
Yes, here it is:
- https://laravel.com/docs/10.x/migrations#renaming-columns
- https://laravel.com/docs/10.x/migrations#modifying-columns
- https://github.com/laravel/docs/pull/8396
- https://github.com/laravel/docs/pull/8456

**Comment by Temepest74** [comment]
> Yes, here it is:
> 
> * https://laravel.com/docs/10.x/migrations#renaming-columns
> * https://laravel.com/docs/10.x/migrations#modifying-columns
> * [[9.x] Remove Doctrine DBAL from prerequisites of renaming/dropping column docs#8396](https://github.com/laravel/docs/pull/8396)
> * [[10.x] Support native column modifying docs#8456](https://github.com/laravel/docs/pull/8456)

Hmm, they should have noted this inside the 9 to 10 upgrade guide :/ Thank you
